dsa-23au/java-dsa/Torsten-David.app/src/main/java/com/ActivityTor/app/AppleWatchDataApp.java

```diff
@@ -71,8 +73,23 @@ public void loadData(){
     public void showAllData(){
         watchData.printAllData(); // Print processed data
     }
-    
-}
 
-    
+    //Method to add data to csv
+    public void appendToCSV(String date, int steps, double distance, int flights, double calories, double handWashing, double restingEnergy, double soundLevel) 
+    {
+        String filePath = "resources/AppleWatchData_myData.csv";
+
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filePath, true))) {
```

Comment:
It looks like this will try to write to a CSV file outside the JAR file. it will look for a directory called "resources" in the same path where you started your JAR file, and if it can't find it, will throw an error.

You can run it this, just create the "resources" directory, or more likely, point it to the Maven resources directory `src/main/resources/data.csv`

```
src
  |--main
      |--java
          |--com
               |--myapp
                   |--App.java
      |--resources
          |--data.csv
```

It will overwrite / append to your CSV file, and you can run `mvn package` again to incorporate a new JAR file with the updated CSV. You won't lose the old CSV file because it's tracked by git, so you can discard the new CSV data changes if you wish, or add / commit them to make them the current version and push it to `origin`.